services/getstream/followTopic.js

```diff
@@ -86,19 +89,37 @@ const addTopicToChatTab = async (token, topicName, userId, isAnonymous) => {
     });
 
     const client = new StreamChat(Environment.GETSTREAM_API_KEY, Environment.GETSTREAM_API_SECRET);
-    await client.connectUser({id: userId}, token);
     const channel = await client.channel('topics', `topic_${topic}`, {
       name: `#${topic}`,
       members: [userId],
       channel_type: 3
     });
 
     console.log('prepare follow channel', topic);
+    let textOwnUser = 'You joined this community';
+    if (isAnonymous) textOwnUser += ' incognito';
+
     await channel.create();
     await channel.addMembers([userId]);
-    await channel.sendMessage({text}, {skip_push: true});
+    if (withSystemMessage) {
+      console.log('system message sent');
+      await channel.sendMessage(
+        {
+          user_id: userId,
+          text: textOwnUser,
+          other_text: '',
+          own_text: textOwnUser,
+          system_user: userId,
+          isSystem: true,
+          type: 'system',
+          only_show_to_system_user: true
+        },
+        {
+          skip_push: true
+        }
+      );
+    }
     console.log('channel followed');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The removal of `client.connectUser` call might cause issues if the user is not already connected. Please ensure that the user connection is handled elsewhere in the code before this function is called. If not, you may need to reintroduce the `client.connectUser` call.

```diff
+    await client.connectUser({id: userId}, token);
     const channel = await client.channel('topics', `topic_${topic}`, {
       name: `#${topic}`,
       members: [userId],
       channel_type: 3
     });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->